### ISSUE TITLE
Build failure: nodePackages.intelephense on Darwin

### ISSUE THREAD

**Issue by ThoFrank** [NixOS/nixpkgs#296141]
### Steps To Reproduce

1. build nodePackages.intelephense from 23.11 or unstable

### Build log

https://gist.github.com/ThoFrank/7335560e12cffe949edd253d4f875763

```
npm ERR! In file included from ../src/GcProfiler.cc:1:
npm ERR! In file included from ../src/GcProfiler.hh:3:
npm ERR! In file included from ../../nan/nan.h:62:
npm ERR! In file included from /nix/store/14a875gw99lnjdwcxibxhbqa60mjja3v-node-sources/src/node.h:73:
npm ERR! In file included from /nix/store/14a875gw99lnjdwcxibxhbqa60mjja3v-node-sources/deps/v8/include/v8.h:21:
npm ERR! In file included from /nix/store/x00khr8pk9w1qxa629b2crwqiw51p3m5-libcxx-16.0.6-dev/include/c++/v1/memory:884:
npm ERR! In file included from /nix/store/x00khr8pk9w1qxa629b2crwqiw51p3m5-libcxx-16.0.6-dev/include/c++/v1/__memory/allocate_at_least.h:13:
npm ERR! In file included from /nix/store/x00khr8pk9w1qxa629b2crwqiw51p3m5-libcxx-16.0.6-dev/include/c++/v1/__memory/allocator_traits.h:14:
npm ERR! In file included from /nix/store/x00khr8pk9w1qxa629b2crwqiw51p3m5-libcxx-16.0.6-dev/include/c++/v1/__memory/construct_at.h:23:
npm ERR! /nix/store/x00khr8pk9w1qxa629b2crwqiw51p3m5-libcxx-16.0.6-dev/include/c++/v1/new:355:14: error: no member named 'aligned_alloc' in the global namespace
npm ERR!     return ::aligned_alloc(__alignment, __size > __rounded_size ? __size : __rounded_size);
npm ERR!            ~~^
npm ERR! 1 error generated.
```

### Additional context

Test

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 23.2.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixpkgs"`
 - channels(thomas): `"darwin, home-manager-23.05.tar.gz, nixpkgs-23.05, unstable"`
 - nixpkgs: `/Users/thomas/.nix-defexpr/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by pineapplehunter** [comment]
This seems to be fixed on nixos-unstable branch after this PR #338761.

**Comment by ThoFrank** [comment]
@pineapplehunter yes thanks for the hint :)
